projects/indicators/src/modules/text-highlight/text-highlight.directive.spec.ts

```diff
@@ -100,6 +100,62 @@ describe('Text Highlight', () => {
     expect(mark.innerHTML.trim()).toBe('text');
   });
 
+  it('should handle empty strings and empty arrays', () => {
+    component.searchTerm = '';
+    fixture.detectChanges();
+
+    expect(containerEl.querySelector('mark')).toBeFalsy();
+
+    component.searchTerm = [];
+    fixture.detectChanges();
+
+    expect(containerEl.querySelector('mark')).toBeFalsy();
+  });
+
+  it('should highlight search terms when using an array', () => {
+    component.searchTerm = ['Here', 'some', 'text'];
+    fixture.detectChanges();
+
+    const marks: NodeListOf<HTMLElement> =
+      fixture.nativeElement.querySelectorAll('mark');
+    expect(marks.length).toEqual(3);
+    expect(marks[0].innerHTML.trim()).toBe('Here');
```

Comment:
Since you're executing this same type of logic over and over again in this file, I'd make a function for it:

```
function validateMark(markEl: HTMLElement, expectedText: string): void {
  expect(markEl.innerHTML.trim()).toBe(expectedText);
}
```

Reply:
Done.